src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp

```diff
@@ -222,7 +257,8 @@ void EmitSetAttribute(EmitContext& ctx, IR::Attribute attr, Id value, u32 elemen
         return;
     }
     const Id pointer{OutputAttrPointer(ctx, attr, element)};
-    ctx.OpStore(pointer, ctx.OpBitcast(ctx.F32[1], value));
+    const Id component_type{OutputAttrComponentType(ctx, attr)};
+    ctx.OpStore(pointer, ctx.OpBitcast(component_type, value));
```

Comment:
Can this check if MRT number format is integer too so it doesn't emit an unnecessary bitcast on the most common case of float

Reply:
This was doing a bitcast to float previously, is it okay to skip that?

Reply:
From my understanding this bitcast is there to ensure type of value written matches the number type of destination MRT. SetAttribute always receives a float input so if MRT is also float which is most common case we can skip the bitcast. Though I guess this is fine because MRT outputs are not very common

Reply:
Okay, updated to skip the bitcast for float.